frontend/src/modules/S3_Datasets/components/DatasetListItem.js

```diff
@@ -2,29 +2,42 @@ import {
   Box,
   Button,
   Card,
+  CircularProgress,
```

Comment:
I do not think it makes sense to have request access button + modal for dataset list page. For dataset list, only datasets which the current user's team is admin / steward to or already shared will show

It is true there are cases where this user then needs to share it to another team or a consumption role that they are a part of or manage respectively - but I think it is more straightforward to keep the `Learn More` from the List View 

And then have the added Request Access Button for all to see on the `DatasetView` page (as you have done so already)